python/arcticdb/version_store/processing.py

```diff
@@ -230,15 +230,19 @@ def value_list_from_args(*args):
         collection = args
     array_list = []
     value_set = set()
+    contains_integer = False
     if len(collection) > 0:
         for value in collection:
             if value not in value_set:
                 value_set.add(value)
                 if isinstance(value, supported_time_types):
                     value = int(value.timestamp() * 1_000_000_000)
-                dtype = np.min_scalar_type(value) if isinstance(value, (int, np.integer)) else None
+                dtype = np.array(value).dtype if isinstance(value, (int, np.integer)) else None
                 array_list.append(np.full(1, value, dtype=dtype))
+                contains_integer = contains_integer or isinstance(value, (int, np.integer))
         value_list = np.concatenate(array_list)
```

Comment:
I wonder if we can use the `casting` argument where available.

The documentation: https://numpy.org/neps/nep-0042-new-dtypes.html#casting

Reply:
that's just a logical operation. is instance is returning a boolean. not sure about your point

Reply:
My comment is on line 243.

Reply:
As in find a way to control whether it "promotes" to float? That'd be neat